agents/dqn.py

```diff
@@ -75,17 +75,20 @@ def forward(self, inputs: Dict[str, Any]) -> Dict[str, Any]:
         # Shape: [batch_size]
         destination_node_idx = inputs[self._destination_node_idx_prefix]
         # TensorWithMask
+        # Shape: [batch_size, max_neighbors_num]
         neighbor_node_ids = inputs[self._neighbors_node_ids_prefix]
 
+        # Shape: [batch_size, max_neighbors_num]
         next_neighbor_q = self._q_network(
             current_node_idx=current_node_idx,
             neighbor_node_ids=neighbor_node_ids,
             destination_node_idx=destination_node_idx
         )
 
-        next_neighbor_ids = neighbor_node_ids.flatten_values[torch.argmax(next_neighbor_q, dim=1)]
-        next_neighbors_idx = _with_random_research(
-            next_neighbor_ids,
+        # Shape: [batch_size]
+        best_next_neighbor_idx = neighbor_node_ids.padded_values[torch.argmax(next_neighbor_q, dim=-1)]
```

Comment:
What if `torch.argmax(next_neighbor_q, dim=-1)` will give you padded idx? Can you please explicitly make something like this `neighbor_node_ids.padded_values[neighbor_node_ids.mask] = -torch.inf` and only then use argmax